WebRoot/js/zimbraAdmin/globalconfig/view/GlobalConfigXFormView.js

```diff
@@ -385,6 +389,22 @@ GlobalConfigXFormView.wipeDeviceListener = function() {
     GlobalConfigXFormView.updateRegisteredDeviceList(obj, result, form);
 }
 
+GlobalConfigXFormView.wipeCancelDeviceListener = function() {
+    var form = this.getForm();
+    var obj = this.getForm().getInstanceValue(ZaGlobalConfig.A2_registeredDevice_Selection)[0];
+    var result = ZaRegisterDevice.wipeCancelDevice(obj)[0];
+
+    GlobalConfigXFormView.updateRegisteredDeviceList(obj, result, form);
+}
+
+GlobalConfigXFormView.blockDeviceListener = function() {
+    var form = this.getForm();
+    var obj = this.getForm().getInstanceValue(ZaGlobalConfig.A2_registeredDevice_Selection)[0];
+    var result = ZaRegisterDevice.blockDevice(obj)[0];
+
+    GlobalConfigXFormView.updateRegisteredDeviceList(obj, result, form);
```

Comment:
most of the code are similar in these functions. can't we just create a common helper function instead?

Reply:
ok